catalog/transformer/catalog-transformer-generic-lib/src/main/java/ddf/catalog/transformer/generic/xml/lib/SaxEventHandlerDelegate.java

```diff
@@ -0,0 +1,378 @@
+    }
+
+    /**
+     * Takes in a sax event from {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate#parser}
+     * and passes it to the {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate#eventHandlers}
+     *
+     * @param namespaceURI the namespaceURI that is passed in by {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate}
+     * @param localName    the localName that is passed in by {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate}
+     * @param qName        the qName that is passed in by {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate}
+     * @throws SAXException
+     */
+    @Override
+    public void endElement(String namespaceURI, String localName, String qName)
+            throws SAXException {
+
+        for (SaxEventHandler transformer : eventHandlers) {
+            transformer.endElement(namespaceURI, localName, qName);
+        }
+
+    }
+
+    /**
+     * Takes in a sax event from {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate#parser}
+     * and passes it to the {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate#eventHandlers}
+     *
+     * @param prefix the prefix that is passed in by {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate}
+     * @param uri    the uri that is passed in by {@link ddf.catalog.transformer.generic.xml.lib.SaxEventHandlerDelegate}
+     * @throws SAXException
+     */
+    @Override
+    public void startPrefixMapping(String prefix, String uri) throws SAXException {
+        for (SaxEventHandler transformer : eventHandlers) {
+            transformer.startPrefixMapping(prefix, uri);
+        }
+    }
+
+    private void configureEventHandlerLookup() {
+
+        for (SaxEventHandler eventHandler : eventHandlers) {
+            if (eventHandler.getWatchedElements() != null) {
+                for (String element : eventHandler.getWatchedElements()) {
+                    List<SaxEventHandler> tmpList = eventHandlerLookup.get(element) != null ?
+                            eventHandlerLookup.get(element) :
+                            new ArrayList<>();
+                    tmpList.add(eventHandler);
+                    eventHandlerLookup.put(element, tmpList);
+                }
+            }
+        }
+
+    }
+
+    public SaxEventHandlerDelegate setMetacardType(MetacardType metacardType) {
+        this.metacardType = metacardType;
+        return this;
+    }
+
+    public TeeInputStream getMetadataStream(InputStream inputStream, OutputStream outputStream) {
+        return new TeeInputStream(inputStream, outputStream);
+    }
+
+}
+
+/**
+ * A private class used to handle errors that occur during SAX Parsing. It allows all the errors
+ * that occur during the parsing of a single document to be easily and succinctly logged at the end
+ * of parsing.
+ */
+class InputTransformerErrorHandler implements ErrorHandler {
```

Comment:
I think it needs to be instantiated and not static because it holds specific information about each parse call. It holds the errors and warnings in its `outWriter` which is then written to `LOGGER.warn` at the end of the parse, in order to not flood the log.